### ISSUE TITLE
Teleport Fails When Target Account Does Not Exist Due to Insufficient Existential Deposit

### ISSUE THREAD

**Issue by asiniscalchi** [paritytech/polkadot-sdk#6198]
When attempting to teleport a native asset from a parachain to AssetHub, the operation fails if the target account does not exist on AssetHub. This is because the native asset is not a **sufficient asset** and cannot be used to satisfy the **Existential Deposit (ED)** required to create a new account. Consequently, the teleportation fails, and the assets remain locked.

## Steps to Reproduce

1. On a parachain configure teleportation of the native asset to AssetHub.
2. Ensure that the target account on AssetHub does not exist (i.e., it has no prior balance).
3. Attempt to teleport the native asset to the non-existent account on AssetHub.

## Expected Behavior

- The teleport operation should succeed by automatically creating the target account on AssetHub.
- If the native asset is insufficient to cover the ED, the system should automatically swap a portion of the teleported assets for a sufficient asset (e.g., DOT) to satisfy the ED and fees.

## Actual Behavior

- The teleport operation fails because the target account does not exist and the native asset cannot be used to pay the ED.
- There is no automatic mechanism to swap teleported assets for a sufficient asset to cover the ED, leading to locked assets.

## Possible Solution

- Implement an automatic process that checks if the target account's balance is below the ED and, if so, swaps a portion of the teleported assets for a sufficient asset to cover the ED.

## Additional Context

- Currently, the lack of automatic account creation when teleporting non-sufficient assets limits the usability of teleportation for native assets.
- Implementing this feature would not only resolve the issue but also be economically advantageous for DOT, as users would need to swap more assets.

**Comment by acatangiu** [comment]
I am personally in favor of doing this, but it has been [discussed before](https://github.com/polkadot-fellows/RFCs/pull/11) without reaching consensus.

@joepetrowski it doesn't look like [wallets/UIs are doing this](https://github.com/polkadot-fellows/RFCs/pull/11#issuecomment-1895786499), what do you think of reopening the discussion of doing it automatically on-chain?

**Comment by joepetrowski** [comment]
Yeah, ideally it's handled off-chain but if that's not happening, maybe doing it automatically is better. It should just come with either an option to opt-in via parameters and/or set a cap of how much you're willing to convert to cover the ED so that people don't get rugged by a changing rate. I'm not sure if that's possible without breaking the XCM format version?

**Comment by eskimor** [comment]
My two cents: 

1. Doing the right thing for UIs should be made bluntly obvious via the provided APIs.
2. We can either fail gracefully (no locked/lost funds) or we make sure we never fail.

**Comment by burdges** [comment]
We've many reasons why UIs should always access RPC nodes for all involved parachains.

We could make teleports require a Merkle proof that the reciever account existed in the recent past.  We could even require this prove some sufficieny lock valid for a few more minutes.  Alone existance does not ensure success without the sufficieny lock, but even without the sufficieny lock this avoids all but the most aggressively bug seeking users.  I suppose sufficieny locks could be implemented within the UI too, but that's still not a guarantee

All that sounds like off-chain messaging, which I suppose makes sense, in that on-chain messaging has never made sense in the first place.
